### PR TITLE
Fix clippy lints

### PR DESCRIPTION
**Pull request by simonsan**
API-Breaking change:
Rename `to_string_lang` -> `get_string_from_lang` due to taking `&self` which is not conventional for `to_*`-functions

r? @the-eater @goto-bus-stop

### CODE REVIEW

File: crates/genie-dat/src/tech_tree.rs [hunk 1]
```diff
@@ -194,6 +194,8 @@ impl TryFrom<i32> for TechTreeDependencyType {
     }
 }
 
+// FIXME: implementation of `From` is preferred since it gives you `Into<_>`
+// for free where the reverse isn't true
```
Comment (the-eater) [inline crates/genie-dat/src/tech_tree.rs:198]: Does something break by replacing this with a `From`?

File: crates/genie-hki/src/lib.rs [hunk 2]
```diff
@@ -917,7 +917,7 @@ impl HotkeyGroup {
 
     /// Returns a string representation of this hotkey group, using the strings
     /// from `lang_file` and the group name string key `sk`.
-    pub fn to_string_lang(&self, lang_file: &LangFile, sk: &StringKey) -> String {
+    pub fn return_string_from_lang(&self, lang_file: &LangFile, sk: &StringKey) -> String {
```
Comment (the-eater) [inline crates/genie-hki/src/lib.rs]: `get_string_from_lang`?

**Review by the-eater — COMMENTED** [review]
Looks good!

could you maybe split the clippy fixes and package updates into separate commits?

**Comment by simonsan** [comment]
> could you maybe split the clippy fixes and package updates into separate commits?

Done

File: crates/genie-drs/src/lib.rs [hunk 3]
```diff
@@ -171,9 +171,9 @@ impl DRSHeader {
     #[inline]
     /// Read a DRS archive header from a `Read`able handle.
     fn from<R: Read>(source: &mut R) -> Result<DRSHeader, Error> {
-        let mut banner_msg = [0 as u8; 40];
-        let mut version = [0 as u8; 4];
-        let mut password = [0 as u8; 12];
+        let mut banner_msg = [0_u8; 40];
+        let mut version = [0_u8; 4];
+        let mut password = [0_u8; 12];
```
Comment (the-eater) [inline crates/genie-drs/src/lib.rs:176]: is `0_u8` a personal preference over `0u8`?

Comment (simonsan) [reply crates/genie-drs/src/lib.rs:176]: imho it's better visible in the source code with an underscore. and let's number part and type easily being distinguished from one another. not sure if only personal preference though.

**Pull request merged**